app/src/main/java/org/fossasia/openevent/api/APIClient.java

```diff
@@ -37,11 +44,61 @@ public APIClient() {
+                            try {
+                                Log.d("Type", "ConversionError");
+                                ErrorDialogEvent errEvent = new ErrorDialogEvent("Conversion Error", String.valueOf(cause.getCause()));
+                                OpenEventApp.postEventOnUIThread(errEvent);
+                                Log.d("ErrorType", String.valueOf(cause.getResponse().getStatus()));
+                            } catch (NullPointerException e) {
+                                Log.d("Exception", "NPE");
+                            }
+                        } else if (cause.getKind().equals(RetrofitError.Kind.HTTP)) {
+                            try {
+                                Log.d("Type", "HTTPError");
+                                ErrorDialogEvent errEvent = new ErrorDialogEvent("HTTP Error", String.valueOf(cause.getResponse().getStatus()));
+                                OpenEventApp.postEventOnUIThread(errEvent);
+                                Log.d("ErrorType", String.valueOf(cause.getResponse().getStatus()));
+                            } catch (NullPointerException e) {
+                                Log.d("Exception", "NPE");
+                            }
+                        } else if (cause.getKind().equals(RetrofitError.Kind.UNEXPECTED)) {
+                            try {
+                                Log.d("Type", "UnexpectedError");
+                                ErrorDialogEvent errEvent = new ErrorDialogEvent("Unexpected Error", String.valueOf(cause.getCause()));
+                                OpenEventApp.postEventOnUIThread(errEvent);
+                                Log.d("ErrorType", String.valueOf(cause.getResponse().getStatus()));
+                            } catch (NullPointerException e) {
+                                Log.d("Exception", "NPE");
+                            }
+                        } else if (cause.getKind().equals(RetrofitError.Kind.NETWORK)) {
+                            try {
+                                Log.d("Type", "NetworkError");
+                                ErrorDialogEvent errEvent = new ErrorDialogEvent("Network Error", String.valueOf(cause.getCause()));
+                                OpenEventApp.postEventOnUIThread(errEvent);
+                                Log.d("ErrorType", String.valueOf(cause.getResponse().getStatus()));
+                            } catch (NullPointerException e) {
+                                Log.d("Exception", "NPE");
+                            }
+                        } else {
+                            Log.d("Type", "Other Error");
+                            Log.d("ErrorType", String.valueOf(cause.getResponse().getStatus()));
+                            ErrorDialogEvent errEvent = new ErrorDialogEvent("Other Error", String.valueOf(cause.getCause()));
+                            OpenEventApp.postEventOnUIThread(errEvent);
```

Comment:
I don't think a switch case is beneficial because I've used .equals() instead of ==, and the former is more preferred than the latter. 

I've chained stuff though, dropping a PR soon :)